File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.js
function solution(cacheSize, cities) {
    let cache = []
    let answer = 0
    cities = cities.map((city) => city.toLowerCase())
    
    for (let i=0; i<cities.length; i++) {
        if (cache.includes(cities[i])) {
            answer += 1
            cache.splice(cache.indexOf(cities[i]), 1)
            cache.unshift(cities[i])
        } else {
            answer += 5
            cache.unshift(cities[i])
        }
        
        if (cache.length > cacheSize) {
            cache.pop()
        }
    }
    return answer;
}

File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.py
def solution(cacheSize, cities):
    cache = []
    answer = 0
    
    processed_cities = [city.lower() for city in cities]
    
    for city in processed_cities:
        if city in cache:
            answer += 1
            cache.remove(city)
            cache.insert(0, city)
        else:
            answer += 5
            cache.insert(0, city)
        if len(cache) > cacheSize:
            cache.pop()
    return answer